examples/multi-agent-debate/main.py

```diff
@@ -0,0 +1,391 @@
+import typer
+import json
+import logging
+from rich.prompt import Prompt, Confirm
+from typing import List, Tuple, Optional, Literal, Any
+import langroid as lr
+from langroid.language_models import OpenAIGPTConfig
+from langroid.agent.chat_agent import ChatAgent, ChatAgentConfig
+from langroid.agent.task import Task
+from langroid.agent.tools.orchestration import DoneTool
+from langroid.agent.tools.google_search_tool import GoogleSearchTool
+from langroid.utils.logging import setup_logger
+from langroid.utils.configuration import set_global
+from config import get_base_llm_config, get_global_settings
+from models import SystemMessages, Message
+
+DEFAULT_SYSTEM_MESSAGE_ADDITION = f"""
+            PLEASE PROVIDE REAL AND VERIFIABLE REFERENCES WITH VALID URLS FOR ALL YOUR ARGUMENTS. 
+            DO NOT MAKE UP YOUR OWN SOURCES; ONLY USE SOURCES YOU FIND FROM A WEB SEARCH. 
+            ENSURE THE CONTENT GENERATED IS FROM REAL SOURCES. 
+            DO NOT REPEAT ARGUMENTS THAT HAVE BEEN PREVIOUSLY GENERATED 
+            AND CAN BE SEEN IN THE DEBATE HISTORY PROVIDED
+            """
+FEEDBACK_AGENT_SYSTEM_MESSAGE = f"""
+            You are an expert and experienced judge specializing in Lincoln-Douglas style debates. 
+            Your goal is to evaluate the debate thoroughly based on the following criteria:
+            1. Clash of Values: Assess how well each side upholds their stated value (e.g., justice, morality) 
+               and how effectively they compare and prioritize values.
+            2. Argumentation: Evaluate the clarity, organization, and logical soundness of each side's case structure, 
+               contentions, and supporting evidence.
+            3. Cross-Examination: Judge the effectiveness of questioning and answering during cross-examination.
+            4. Rebuttals: Analyze how well each side refutes their opponent's arguments.
+            5. Persuasion: Assess communication quality, tone, rhetorical effectiveness, and emotional/ethical appeals.
+            6. Technical Execution: Identify if major arguments were addressed or dropped and check consistency.
+            7. Debate Etiquette: Evaluate professionalism, respect, and demeanor.
+            8. Final Focus: Judge the strength of closing speeches, how well they summarize the case, 
+            and justify a winner.
+            Provide constructive feedback for each debater, 
+            summarizing their performance and declaring a winner with justification.
+            """
+GOOGLE_SEARCH_SYSTEM_MESSAGE = f"""
+            You are a helpful assistant. Extract all the links in references, ensure the URLs looks valid.  
+            Use the GoogleSearchTool to validate the references.
```

Comment:
When this tool is enabled, langroid generates instructions using the tool's `request` field as the name of the tool, so you have to say something like 

> use the TOOL `{GoogleSearchTool.name()}` ...